rfcs/0018-glides-trainsheet-events.md

```diff
@@ -0,0 +1,571 @@
+- `consist` (array of Optional Car, optional, default: empty list): the cars assigned to perform this trip. If a car is `null`, then no car is currently assigned to that position in the consist.
+- `operators` (list of Optional Operator, optional, default: empty list): the array of operators assigned to perform this trip. If an operator is `null`, then no operator is assigned to the car in that position in the consist.
+- `comment` (string, optional): free text with additional information about the trip
+
+*Notes*
+- At least one of `startLocation` and `endLocation` is required.
+- If `startLocation` is 
+- The event should only include values which were explicitly included by the `author`: inferred values should not be included.
+- `consist` and `operators` MUST be the same length; if they are not, the behavior of consumers is undefined.
+
+### Trip dropped
+This removes a trip (either added or scheduled), indicating that it will not be running.
+
+Event type: `com.mbta.ctd.glides.trip-dropped`
+Fields in the event:
+- `author` (Author): the inspector dropping the trip.
+- `tripKey` (Trip Key): which trip to drop. can be either an added or scheduled trip.
+- `reason` (string): a text description of why the trip is being dropped.
+
+### Trip restored
+This restores a trip (either added or scheduled) that was previously dropped.
+
+Event type: `com.mbta.ctd.glides.trip-restored`
+Fields in the event:
+- `author` (Author): the inspector restoring the trip.
+- `tripKey` (Trip Key): which trip to restore. can be either an added or scheduled trip.
+
+### Trip updated
+This indicates that a trip was updated in some fashion:  consist, operators, departure time. This includes updates which confirm an arrival or departure time. Fields which are not present are not considered to be updated.
+
+Event type: `com.mbta.ctd.glides.trip-updated`
+Fields in the event:
+- `author` (Author): the inspector adding the trip
+- `tripKey` (Trip Key): which trip is being updated
+- `comment` (string, optional): free text information about the trip
+- `route` (string, optional): if specified, the new route tag for the trip. This is different from a GTFS route ID.
+- `location` (Location, optional): where the time/consist is being updated for. This can be different from the location in the `author` value or in the `tripKey`, if the author is updating information about the train relative to a different location.
+- `arrivalTime` (Time, optional): if present, the new time that the train is expected to arrive at the given location. If `location` is not specified, this is the time the vehicle is arriving at the final location of the trip.
+- `departureTime` (Time, optional): if present, the new time that the train is expected to depart the given location. If `location` is not specified, this is the time the vehicle is departing the first location of the trip.
```

Comment:
These should explicitly inherit from `author.location` as well as `location`.

Reply:
I've updated this to make `location` required if either of the times are being set, rather than inferring the location based on the author or the trip origin/destination.